plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -2181,12 +2181,13 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
         HiveTableHandle hiveTable = (HiveTableHandle) table;
 
         List<ColumnHandle> partitionColumns = ImmutableList.copyOf(hiveTable.getPartitionColumns());
-        List<HivePartition> partitions = partitionManager.getOrLoadPartitions(metastore, new HiveIdentity(session), hiveTable);
-
-        TupleDomain<ColumnHandle> predicate = createPredicate(partitionColumns, partitions);
 
+        TupleDomain<ColumnHandle> predicate = TupleDomain.all();
         Optional<DiscretePredicates> discretePredicates = Optional.empty();
-        if (!partitionColumns.isEmpty()) {
+        if (!partitionColumns.isEmpty() && hiveTable.getPartitions().isPresent()) {
```

Comment:
Does it mean that we will only be able to get properties after any predicate was pushed (even `AddExchanges` won't cause partitioning to be extracted)?
Does predicate have to be on partitioned column for `hiveTable.getPartitions()` to be present?

IMO we should enumarate partitions up to some limit. Otherwise we will loose some valuable predicate information which could cause perf regressions

Reply:
We should get it per:
```java
        HivePartitionResult partitionResult = partitionManager.getPartitions(metastore, new HiveIdentity(session), handle, constraint);
        HiveTableHandle newHandle = partitionManager.applyPartitionResult(handle, partitionResult, constraint.getPredicateColumns());
```

from `applyFilter`.

Unless constraint does not filter on partition columns.

Reply:
> IMO we should enumarate partitions up to some limit. Otherwise we will loose some valuable predicate information which could cause perf regressions

Would that be correct?

Reply:
> Would that be correct?

You enumerate first 1000 partitions (if they are not loaded). If there are more, you fallback to `all`? Can this be done?